infra/modules/azure_function_app/tier.tf

```
@@ -0,0 +1,37 @@
+## Tier Definitions
+# Centralized mapping for legacy and new tiers
+
+locals {
+  # Tier mappings for better scalability and readability
+  tier_map = {
+    "test"     = "s"
+    "standard" = "m"
+    "premium"  = "l"
+    "xs"       = "xs"
+    "s"        = "s"
+    "m"        = "m"
+    "l"        = "l"
+    "xl"       = "xl"
+  }
+
+  # Default to the mapped tier
+  tier = local.tier_map[var.tier]
+
+  # SKU name mapping
+  sku_name_mapping = {
+    "xs" = "B1"
+    "s"  = "B1"
+    "m"  = "P0v3"
+    "l"  = "P1v3"
+    "xl" = "P2mv3"
+  }
+
+  # Worker process count based on tier
+  worker_process_count_mapping = {
+    "xs" = 2
+    "s"  = 2
```

```suggestion
    "s"  = 1
```
B1 has 1.75GB, so this shall be 1 (1 worker x 4GB=4FB)